gdal/frmts/tiledb/tiledbdataset.cpp

```diff
@@ -315,6 +319,51 @@ CPLErr TileDBRasterBand::IWriteBlock( CPL_UNUSED int nBlockXOff,
                && nBlockYOff >= 0
                && pImage != nullptr );
 
+    if ( poGDS->lpoAttributeDS.size() > 0 )
+    {
+        for ( auto const& poAttrDS: poGDS->lpoAttributeDS )
+        {
+            GDALRasterBand* poAttrBand = poAttrDS->GetRasterBand( nBand );
+            GDALDataType eAttrType = poAttrBand->GetRasterDataType();
+            int nBytes = GDALGetDataTypeSizeBytes( eAttrType );
+            int nValues = nBlockXSize * nBlockYSize;
+            void* pAttrBlock = CPLMalloc( nBytes * nValues );
```

Comment:
use VSIMalloc() and test return value

Reply:
where is pAttrBlock freed ?

Reply:
my fault, I missed that. I have added `CPLFree`